### PR TITLE
registry: add support for extended repository names (PROJQUAY-1535)

### PR DESCRIPTION
**Pull request by kleesc**
Allows forward slashes to be used in repository names according to
https://docs.docker.com/docker-hub/repos/.

NOTE: This change simply allows the use of "/" in repository
names needed for certain Openshift use cases. This does not implement
any new permission model for nested paths. i.e A repository with a
nested path is treated as a single repository under a _single_
namespace.

### CODE REVIEW

File: util/names.py [hunk 1]
```diff
@@ -6,7 +6,10 @@
 from uuid import uuid4
 
 REPOSITORY_NAME_REGEX = re.compile(r"^[a-z0-9][\.a-z0-9_-]{0,254}$")
-
+# Extended repostitory name regex: allows "/" in repo names
+REPOSITORY_NAME_EXTENDED_REGEX = re.compile(
+    "^(?=.{0,255}$)[a-z0-9][\.a-z0-9_-]*(?:\/[a-z0-9][\.a-z0-9_-]*)*$"
```
Comment (BillDett) [inline util/names.py:11]: @kleesc - should this be a raw string (r"...") as well like the REPOSITORY_NAME_REGEX?

**Comment by kleesc** [comment]
@jonathankingfc @thomasmckay Ping

File: requirements.txt [hunk 2]
```diff
@@ -169,4 +169,3 @@ xhtml2pdf==0.2.4
 xmltodict==0.12.0
 yapf==0.29.0
 zipp==2.1.0
-zope.interface==4.7.1
```
Comment (thomasmckay) [inline requirements.txt:172]: Remove from requirements-osbs.txt as well

**Review by thomasmckay — CHANGES_REQUESTED** [review]

**Review by jonathankingfc — APPROVED** [review]

**Review by jonathankingfc — APPROVED** [review]

**Review by jonathankingfc — APPROVED** [review]

**Review by thomasmckay — APPROVED** [review]
LGTM

**Pull request merged**